core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -122,7 +122,12 @@ private boolean purgeInMemoryContents(LocalizedCacheTopology topology) {
          BlockingQueue<CompletableFuture<?>> expirationPermits = new ArrayBlockingQueue<>(MAX_CONCURRENT_EXPIRATIONS);
          long currentTimeMillis = timeService.wallClockTime();
 
-         IntSet segments = IntSets.from(topology.getReadConsistentHash().getPrimarySegmentsForOwner(localAddress));
+         IntSet segments;
+         if (topology.getMembersSet().contains(localAddress)) {
```

Comment:
I am just curious what happens if this isn't done? It seems a bit out of scope of this issue, but maybe it is needed.

Reply:
There's an annoying warning: https://issues.redhat.com/browse/ISPN-10882